src/nvim/api/options.c

```diff
@@ -309,15 +309,55 @@ Dictionary nvim_get_all_options_info(Error *err)
 ///     - commalist: List of comma separated values
 ///     - flaglist: List of single char flags
 ///
+/// The last set information applies to the local value in the current buffer
+/// or window if it is available, otherwise the global value information is
+/// returned.
 ///
 /// @param          name Option name
 /// @param[out] err Error details, if any
 /// @return         Option Information
 Dictionary nvim_get_option_info(String name, Error *err)
   FUNC_API_SINCE(7)
 {
-  return get_vimoption(name, err);
+  return get_vimoption(name, curbuf, curwin, err);
```

Comment:
This is needed to fix the linked issue. Not sure though if it is a good idea to change the behavior of this existing function. Alternatively we could pass NULLs here instead of both curbuf and curwin.

Reply:
Since `nvim_get_option_info` is deprecated now, I changed it back to the old behavior - it always uses the global value.